rfc/src/rfcs/0002-function-stubbing.md

```diff
@@ -156,7 +156,45 @@ Given a set of `original`-`replacement` pairs, Kani will exit with an error if
 1. a specified `original` function/method does not exist;
 2. a specified `replacement` stub does not exist;
 3. the user specifies conflicting stubs for the same harness (e.g., if the same `original` function is mapped to multiple `replacement` functions); or
-4. the signature of the `replacement` stub is not compatible with the signature of the `original` function/method.
+4. the signature of the `replacement` stub is not compatible with the signature of the `original` function/method (see next section).
+
+### Stub compatibility and validation
+
+When considering whether a function/method can be replaced with some given stub, we want to allow some measure of flexibility, while also ensuring that we can provide the user with useful feedback if stubbing results in misformed code.
+We consider a stub to be compatible with some function/method if all the following conditions are met:
+
+- They have the same return type. The names of type variables need to match.
+- They have the same number of parameters.
+- Each parameter in the stub has the same type as the corresponding parameter in the original function/method.
+The names of type variables need to match.
+- During monomorphization, every statically dispatched call to a trait method can be resolved.
```

Comment:
Can you please document this in terms of the language not the compiler phase? Something like: "The bounds for each type parameter don't need to match; however, all calls to the original function must also satisfy the bounds of the stub."

Reply:
Good idea - updated to this language